datahub_sap_hana/ingestion.py

```diff
@@ -1,40 +1,216 @@
-from typing import Any, Dict
-
+import logging
+from collections import defaultdict
+from typing import Any, Dict, Iterable, List, Tuple, Union
 import sqlalchemy_hana.types as custom_types  # type: ignore
+from datahub.configuration.common import AllowDenyPattern
+from datahub.emitter import mce_builder
+from datahub.emitter.mcp_builder import mcps_from_mce
 from datahub.ingestion.api.common import PipelineContext
 from datahub.ingestion.api.decorators import (
     config_class,  # type: ignore
     platform_name,  # type: ignore
 )
+from datahub.ingestion.api.workunit import MetadataWorkUnit
 from datahub.ingestion.source.sql.sql_common import (
     SQLAlchemySource,
-    register_custom_type,  # type: ignore
+    SqlWorkUnit,
+    register_custom_type,
 )
 from datahub.ingestion.source.sql.sql_config import BasicSQLAlchemyConfig
 from datahub.metadata.com.linkedin.pegasus2avro import schema
+from pydantic import BaseModel, Field
+from sqlalchemy import create_engine
 
 register_custom_type(custom_types.TINYINT, schema.NumberType)
 
 
+logger: logging.Logger = logging.getLogger(__name__)
+
+# This query only takes 4 columns to match the fields in the ViewLineageEntry class. It also ignores schemas that contain SYS
+# Object dependencies in SAP HANA https://help.sap.com/docs/SAP_HANA_PLATFORM/de2486ee947e43e684d39702027f8a94/5ce9a6584eb84f10afbbf2b133534932.html
+LINEAGE_QUERY = """
+SELECT 
+    BASE_OBJECT_NAME as source_table, 
+    BASE_SCHEMA_NAME as source_schema,
+    DEPENDENT_OBJECT_NAME as dependent_view, 
+    DEPENDENT_SCHEMA_NAME as dependent_schema
+  from SYS.OBJECT_DEPENDENCIES 
+WHERE 
+  DEPENDENT_OBJECT_TYPE = 'TABLE'
+  OR DEPENDENT_OBJECT_TYPE = 'VIEW'
+  AND BASE_SCHEMA_NAME NOT LIKE '%SYS%'
```

Comment:
Actually, if we use `*SYS*` as the default for the allow/deny pattern for schemas, then we don't need any filtering on the base_schema either